packages/arb-ts/src/lib/message/L1Transaction.ts

```diff
@@ -0,0 +1,213 @@
+      InboxMessageDeliveredFromOrigin: iface.getEventTopic(
+        messageDeliveredFromOrigin
+      ),
+    }
+    const logs = this.logs.filter(
+      log =>
+        log.topics[0] === eventTopics.InboxMessageDelivered ||
+        log.topics[0] === eventTopics.InboxMessageDeliveredFromOrigin
+    )
+    return logs.map(log => BigNumber.from(log.topics[1]))
+  }
+
+  /**
+   * Get any l1tol2 messages created by this transaction
+   * @param l2SignerOrProvider
+   */
+  public async getL1ToL2Messages<T extends SignerOrProvider>(
+    l2SignerOrProvider: T
+  ): Promise<L1ToL2MessageReaderOrWriter<T>[]>
+  public async getL1ToL2Messages<T extends SignerOrProvider>(
+    l2SignerOrProvider: T
+  ): Promise<L1ToL2MessageReader[] | L1ToL2MessageWriter[]> {
+    const provider = SignerProviderUtils.getProviderOrThrow(l2SignerOrProvider)
+
+    const chainID = (await provider.getNetwork()).chainId.toString()
+
+    const messageNumbers = this.getMessageNumbers()
+    if (!messageNumbers || messageNumbers.length === 0) return []
+
+    return messageNumbers.map((mn: BigNumber) => {
+      const ticketCreationHash = L1ToL2Message.calculateRetryableCreationId(
+        BigNumber.from(chainID),
+        mn
+      )
+      return L1ToL2Message.fromRetryableCreationId(
+        l2SignerOrProvider,
+        ticketCreationHash,
+        mn
+      )
+    })
+  }
+
+  /**
+   * Gets a single l1ToL2Message
+   * If the messageIndex is supplied the message at that index will be returned.
+   * If no messageIndex is supplied a message will be returned if this transaction only created one message
+   * All other cases throw an error
+   * @param l2SignerOrProvider
+   */
+  public async getL1ToL2Message<T extends SignerOrProvider>(
+    l2SignerOrProvider: T,
+    messageNumberIndex?: number
+  ): Promise<L1ToL2MessageReaderOrWriter<T>>
+  public async getL1ToL2Message<T extends SignerOrProvider>(
+    l2SignerOrProvider: T,
+    messageIndex?: number
+  ): Promise<L1ToL2MessageReader | L1ToL2MessageWriter> {
+    const allL1ToL2Messages = await this.getL1ToL2Messages(l2SignerOrProvider)
+    const messageCount = allL1ToL2Messages.length
+    if (!messageCount)
+      throw new ArbTsError(
+        `No l1 to L2 message found for ${this.transactionHash}`
+      )
+
+    if (messageIndex !== undefined && messageIndex >= messageCount)
+      throw new ArbTsError(
+        `Provided message number out of range for ${this.transactionHash}; index was ${messageIndex}, but only ${messageCount} messages`
+      )
+    if (messageIndex === undefined && messageCount > 1)
+      throw new ArbTsError(
+        `${messageCount} L2 messages for ${this.transactionHash}; must provide messageNumberIndex (or use (signersAndProviders, l1Txn))`
+      )
+
+    return allL1ToL2Messages[messageIndex || 0]
+  }
+
+  /**
+   * Get any deposit events created by this transaction
+   * @returns
+   */
+  public getDepositEvents(): DepositInitiatedEvent['args'][] {
```

Comment:
I think we should make it more explicit that this refers to an asset deposit
```suggestion
  public getTokenDepositEvents(): DepositInitiatedEvent['args'][] {
```

Reply:
Updated